app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt

```diff
@@ -78,61 +143,60 @@ class EditPatientScreen(context: Context, attributeSet: AttributeSet) : Relative
 
     TheActivity.component.inject(this)
 
-    val screenDestroys = RxView.detaches(this).map { ScreenDestroyed() }
-
-    bindUiToController(
-        ui = this,
-        events = Observable.mergeArray(
-            screenCreates(),
-            saveClicks(),
-            nameTextChanges(),
-            phoneNumberTextChanges(),
-            districtTextChanges(),
-            stateTextChanges(),
-            colonyTextChanges(),
-            genderChanges(),
-            dateOfBirthTextChanges(),
-            dateOfBirthFocusChanges(),
-            ageTextChanges(),
-            backClicks()
-        ),
-        controller = controller,
-        screenDestroys = screenDestroys
-    )
+    delegate.prepare()
+
   }
 
-  private fun screenCreates(): Observable<UiEvent> {
-    val key = screenRouter.key<EditPatientScreenKey>(this)
-    val patientEditScreenCreated = ScreenCreated
-        .from(key.patient, key.address, key.phoneNumber)
-    return Observable.just(patientEditScreenCreated)
+  override fun onAttachedToWindow() {
+    super.onAttachedToWindow()
+    delegate.start()
+    eventsDisposable = events.subscribe { delegate.eventSource.notifyEvent(it) }
+  }
+
+  override fun onDetachedFromWindow() {
+    delegate.stop()
+    if (::eventsDisposable.isInitialized && eventsDisposable.isDisposed.not()) {
```

Comment:
I find the way these events are being managed is cumbersome. Can we do something to simplify them? As a suggestion,

- Provide the `Observable<E>` to the `MobiusDelegate` via the constructor
- Subscribe and unsubscribed when `start()` and `stop()` are called

This is a suggestion for a future PR.

Reply:
Yes, can do!